src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -392,6 +391,10 @@ private static unsafe int AlpnServerSelectCallback(IntPtr ssl, out byte* outp, o
                 return Ssl.SSL_TLSEXT_ERR_NOACK;
             }
 
+            // No common application protocol was negotiated, set the target on the alpnHandle to null.
+            // It is ok to clear the handle value here, this results in handshake failure, so the SslStream object is disposed.
+            protocolHandle.Target = null;
```

Comment:
Can just call free at this point. Then you don't need to worry about the free'ing later. Looking at the code it won't matter if it's double free'd and it freeing it will make the target null anyway.

Reply:
> Then you don't need to worry about the free'ing later.

This will anyways get freed, when the context is being disposed, since the isallocated value is not updated when free is called on the copy in matching algo.